dbcon/mysql/ha_mcs_impl.cpp

```diff
@@ -1304,11 +1302,18 @@ uint32_t doUpdateDelete(THD* thd, gp_walk_info& gwi, const std::vector<COND*>& c
             else
                 aliasName = item->table_name.str;
 
+            if (lower_case_table_names)
+            {
+                boost::algorithm::to_lower(aliasName);
+                boost::algorithm::to_lower(tableName);
+                boost::algorithm::to_lower(tmpTableName);
+            }
+
             if (strcasecmp(tableName.c_str(), "") == 0)
             {
                 tableName = tmpTableName;
             }
-            else if (strcasecmp(tableName.c_str(), tmpTableName.c_str()) != 0)
+            else if (strcmp(tableName.c_str(), tmpTableName.c_str()) != 0)
```

Comment:
Shouldn't the sensitivity of the string comparison be based on `lower_case_table_names`? That is:
```
if (lower_case_table_names) strcasecmp()
else strcmp()
```

Reply:
tableName and tmpTableName are converted just above, so the strcasecmp is no longer needed.